libsql-sys/src/connection.rs

```diff
@@ -213,6 +241,28 @@ impl<W: Wal> Connection<W> {
             self.conn
         }
     }
+
+    pub fn db_change_counter(&self) -> Result<u32, std::ffi::c_int> {
+        let mut file_ptr: *mut libsql_ffi::sqlite3_file = std::ptr::null_mut();
+        let rc = unsafe {
+            libsql_ffi::sqlite3_file_control(
+                self.handle(),
+                "main\0".as_ptr() as *const _,
+                libsql_ffi::SQLITE_FCNTL_FILE_POINTER,
+                &mut file_ptr as *mut _ as *mut _,
+            )
+        };
+        if rc != libsql_ffi::SQLITE_OK {
+            return Err(rc);
+        }
+        let counter = unsafe {
+            let mut counter: u32 = 0;
+            let file = &*file_ptr;
+            (*file.pMethods).xRead.unwrap()(file_ptr, &mut counter as *mut _ as *mut _, 4, 24);
+            u32::from_be(counter)
+        };
+        Ok(counter)
+    }
```

Comment:
What's that for?

Reply:
db change counter is a piece of metadata kept in page1 we can use to compare which database file is newer. We used to read it directly from the db, but that was cutting corners, assuming that we don't use any custom VFS. Now we do, so we need to read the value via VFS. Unfortunately there's no pragma or any other way that directly gives you back the change counter, so instead we go through sqlite3_file_control to get the VFS pointer and call its xRead function to read the page1 header.